docs/manual/determinism-checker.tex

```diff
@@ -57,9 +57,10 @@
 \item[\refqualclass{checker/determinism/qual}{NonDet}] indicates
   that the expression might have different values in two different executions on the same machine.
 \item[\refqualclass{checker/determinism/qual}{OrderNonDet}] indicates that
-  a collection (i.e any class that is a subtype of java.util.Collection or java.util.Iterator) or an array will have the same elements in every execution, but in a
+  a collection or an array will have the same elements in every execution, but in a
   possibly different order.  \<@OrderNonDet> may only be written on
-  collections and arrays. An \<@OrderNonDet> Map indicates that the its keySet is \<@OrderNonDet>.
+  types whose declaration is annotated with \refqualclass{checker/determinism/qual}{CollectionType} and arrays.
```

Comment:
Saying "is annotated with CollectionType" is a bit operational.  Let's say the intention, and we can also say how it is implemented (which is via the CollectionType annotation).